packages/xstate-immer/src/index.ts

```diff
@@ -61,3 +63,52 @@ export function createUpdater<TContext, TEvent extends ImmerUpdateEvent>(
     type
   };
 }
+
+const IMMER_PATCH_TYPE = '@immer.patch' as const;
+
+export interface ImmerPatchEvent<K> {
+  type: typeof IMMER_PATCH_TYPE;
+  key: K;
+  patches: Patch[];
+}
+
+export const createPatch = <T, K extends keyof T>(
+  key: K
+): {
+  (currentValue: T[K], fn: (draft: Draft<T[K]>) => void): ImmerPatchEvent<K>;
+  type: ImmerPatchEvent<K>['type'];
+  action: AssignAction<T, ImmerPatchEvent<K>>;
+  transition: Record<
+    ImmerPatchEvent<K>['type'],
+    { actions: Array<AssignAction<T, ImmerPatchEvent<K>>> }
+  >;
+} => {
+  const applyPatch = (
```

Comment:
In https://github.com/davidkpiano/xstate/pull/1129 you have implemented updater as a function with extra properties but after a discussion, it has been changed to be a plain object with properties. So I would say that the same should be done for this new `createPatch`, even if only for the API consistency.